view/src/bin/pviewd.rs

```diff
@@ -77,28 +115,119 @@ async fn main() -> Result<()> {
                 params,
             )
             .await?;
-            Ok(())
-        }
-        Command::Start { host, view_port } => {
-            tracing::info!(?opt.sqlite_path, ?host, ?view_port, ?opt.node, ?opt.pd_port, "starting pviewd");
-
-            let storage = penumbra_view::Storage::load(opt.sqlite_path).await?;
-
-            let service = ViewService::new(storage, opt.node, opt.pd_port).await?;
-
-            tokio::spawn(
-                Server::builder()
-                    .accept_http1(true)
-                    .add_service(tonic_web::enable(ViewProtocolServiceServer::new(service)))
-                    .serve(
-                        format!("{}:{}", host, view_port)
-                            .parse()
-                            .expect("this is a valid address"),
-                    ),
-            )
-            .await??;
+
+            // Create config file
+
+            let kms_config: Option<soft_kms::Config> = match seed_phrase {
+                Some(seed_phrase) => {
+                    let spend_key =
+                        SpendKey::from_seed_phrase(SeedPhrase::from_str(seed_phrase.as_str())?, 0);
+
+                    let pak = ed25519_consensus::SigningKey::new(rand_core::OsRng);
+                    let pvk = pak.verification_key();
+
+                    let auth_policy = vec![
+                        AuthPolicy::OnlyIbcRelay,
+                        AuthPolicy::DestinationAllowList {
+                            allowed_destination_addresses: vec![
+                                spend_key
+                                    .incoming_viewing_key()
+                                    .payment_address(Default::default())
+                                    .0,
+                            ],
+                        },
+                        AuthPolicy::PreAuthorization(PreAuthorizationPolicy::Ed25519 {
+                            required_signatures: 1,
+                            allowed_signers: vec![pvk],
+                        }),
+                    ];
+                    Some(soft_kms::Config {
+                        spend_key: spend_key.clone(),
+                        auth_policy,
+                    })
+                }
```

Comment:
This could probably be factored out into a standalone function that would build a default `soft_kms::Config` from a seed phrase